### PR TITLE
APP-1599 Implement symphony-app-authentication-api

### PR DESCRIPTION
**Pull request by robson-sanchez**
- Moving all the source code have already been implemented in the integration-core to the new repo adapting the code to follow the interfaces created previously.

### CODE REVIEW

File: symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/AppAuthenticationApiClientTest.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+          e.getMessage());
+    }
+
+    verify(client, times(1)).close();
+  }
+
+  @Test
+  public void testRequestOK() throws AppAuthenticationException {
+    doReturn(Response.Status.OK.getStatusCode()).when(response).getStatus();
+    doReturn("{ \"appId\": \"APP_ID\", \"appToken\": \"ABCD\" }").when(response).readEntity(String.class);
```
Comment (crepache) [inline symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/AppAuthenticationApiClientTest.java]: Create a constant.

File: symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/AppAuthenticationServletTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    JsonParser parser = JsonParserFactory.getInstance().getComponent();
+    String expected = parser.writeToString(errorResponse);
+
+    verify(response, times(1)).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    verify(writer, times(1)).write(expected);
+  }
+
+  @Test
+  public void testMissingPodId() throws IOException, ServletException {
+    doReturn("{ \"appId\": \"APP_ID\" }").doReturn(null).when(reader).readLine();
```
Comment (crepache) [inline symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/AppAuthenticationServletTest.java]: Create a constant.

File: symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/TokensValidationServletTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    JsonParser parser = JsonParserFactory.getInstance().getComponent();
+    String expected = parser.writeToString(errorResponse);
+
+    verify(response, times(1)).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    verify(writer, times(1)).write(expected);
+  }
+
+  @Test
+  public void testMissingAppToken() throws IOException, ServletException {
+    doReturn("{ \"appId\": \"APP_ID\" }").doReturn(null).when(reader).readLine();
```
Comment (crepache) [inline symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/TokensValidationServletTest.java]: Create a constat.

File: symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/TokensValidationServletTest.java [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    JsonParser parser = JsonParserFactory.getInstance().getComponent();
+    String expected = parser.writeToString(errorResponse);
+
+    verify(response, times(1)).setStatus(HttpServletResponse.SC_BAD_REQUEST);
+    verify(writer, times(1)).write(expected);
+  }
+
+  @Test
+  public void testMissingSymphonyToken() throws IOException, ServletException {
+    doReturn("{ \"appId\": \"APP_ID\", \"appToken\": \"ABCD\" }").doReturn(null).when(reader).readLine();
```
Comment (crepache) [inline symphony-app-authentication-api/src/test/java/org/symphonyoss/symphony/apps/authentication/servlets/TokensValidationServletTest.java]: Create a constant.

File: symphony-app-authentication-api/pom.xml [hunk 5]
```diff
@@ -28,6 +28,10 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-all</artifactId>
```
Comment (crepache) [inline symphony-app-authentication-api/pom.xml:33]: Add the scope test.

**Review by crepache — CHANGES_REQUESTED** [review]

File: symphony-app-authentication-api/pom.xml (hunk 5 above)
Comment (robson-sanchez) [reply symphony-app-authentication-api/pom.xml:33]: This scope has already been defined in the dependency management.

**Review by crepache — APPROVED** [review]

**Comment by crepache** [comment]
LGTM

**Pull request merged**